### ISSUE TITLE
Use CDN URLs for attachments instead of S3 URLs

### ISSUE THREAD

**Issue by mxk0** [HHS/simpler-grants-gov#3094]
### Summary

Once the CDN is in place and in front of S3, we'll need to update the document links in API responses to reference the CDN address for each document.

### Acceptance criteria

- [ ] API responses for attached documents reference the CDN URL, not a pre-signed S3 URL.

**Comment by mdragon** [comment]
@chouinar Do we want to have 2 columns, a public and private URL? Because we'd still need the proper S3 locations for Posting/Attachment management purposes, so I was thinking we'd just transform these live since the CDN will just be like a prefix replacement for the root of the S3 style URL?

**Comment by chouinar** [comment]
@mdragon 

My understanding (please correct if I'm wrong) is that the way a CDN works in from of s3 is that if a file is in `s3://my_bucket/path/to/file.txt` you can access it at `my-fun-cdn-name/path/to/file.txt`. If so, as you say, we just do a text replace of `s3//my_bucket` with `my-fun-cdn-name` to magically make it work. This is even simpler than the approach for pre-signing URLs, and even better can just be swapped out on the fly without any issue.

**Comment by mikehgrantsgov** [comment]
@chouinar this is correct. One additional note is the CDN can specify a bucket prefix, so objects can be stored in s3 at `s3://my_bucket/cdn/file.txt` and if the CDN is configured to have a `/cdn` prefix then the file would be accessible at `https://cdn_url/file.txt`. Just an option in case it's useful.
